pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -215,6 +218,489 @@ def get_strategic_model_unit_container():
 )
 
 
+# TODO: move this to utility
+class MissingDataError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+# TODO: Move this function to a utility
+def get_trucking_arc_list():
+    return [
+        "PCT",
+        "PKT",
+        "PST",
+        "PRT",
+        "POT",
+        "FCT",
+        "CKT",
+        "CST",
+        "CRT",
+        "CCT",
+        "SCT",
+        "SKT",
+        "SOT",
+        "RKT",
+        "RST",
+        "ROT",
+    ]
+
+
+# TODO: Move this function to a utility
+def get_piping_arc_list():
+    return [
+        "PCA",
+        "PNA",
+        "PPA",
+        "CNA",
+        "CCA",
+        "NNA",
+        "NCA",
+        "NKA",
+        "NSA",
+        "NRA",
+        "NOA",
+        "FCA",
+        "RNA",
+        "RCA",
+        "RKA",
+        "RSA",
+        "SNA",
+        "SCA",
+        "SKA",
+        "SRA",
+        "SOA",
+        "ROA",
+    ]
+
+
+# TODO: Move this function to a utility
+def _check_optional_data(
+    df_sets,
+    df_parameters,
+    optional_tab_name,
```

Comment:
Is this `optional_tab_name` argument meant to always correspond to a set instead of a parameter? If so, I think the argument name should reflect that. Also, I think the word "optional" in the name is confusing here, because the tab name that is passed in this argument is often something which is in fact not optional (e.g., "CompletionsPads").

Reply:
Good note. I updated the name to include "set". I do believe these will all be optional. As of now, we require no Sets in PARETO, we do however require a number of "one of " sets. For example, one of ["ProductionPads", "CompletionsPads", "FreshwaterSources"] (water sources). This can be modified pretty easily though if we want to change what is required.